### DISCUSSION TITLE
Post translational modification

### DISCUSSION THREAD

**Discussion by amiel-ab** [marrink-lab/vermouth-martinize#597] [Ideas]
Hi all, 

I am trying to add post translationnal modification that is a palmitoyl chain to a lysine from a protein, in CG (Martini3).
@fgrunewald advised me to use  lipidation parameters from here https://pubmed.ncbi.nlm.nih.gov/38019969/
However, their lipidation parameters are about cysteine-palmitoyl but not lysine-palmitoyl.

Do you have any idea on how i could use polyply to generate such modified structure? 

Thank you for your consideration.

**Comment by fgrunewald** [comment]
@amiel-ab,

My recommendation would be to build an all-atom model of your lipidation using the CHARMM-GUI. Subsequnetly, we can show you how the martinize2 files can be adopted to generate coordinates and input files for your protein. Please note that for this purpose I will later transfer the discussion from the polyply to the martinize2 GitHub page. 

Just reply to the issue once you obtained the all-atom structure

Cheers,
Fabian

**Reply by amiel-ab** [reply]
Thank you for your reply !
I already got the all atoms structure with the post-translationnal modifications i tried with martinize2 it does'nt work (stay blocked and at some point the process is killed) i know there is a -modify option in martinize2 but i am not familiar with it.
Maybe you have some suggestions ?

**Reply by pckroon** [reply]
Since lipidations are so large you're probably better off defining it as a separate residue rather than as a Modification.
Take a look at https://vermouth-martinize.readthedocs.io/en/latest/tutorials/6_adding_residues_links/index.html to get started :)

**Reply by amiel-ab** [reply]
Thanks a lot @pckroon i'll have a look

**Reply by amiel-ab** [reply]
Hi,
I tried the tutorial
I will do the input .ff block in charmm36 by hand but for the output one i am unsure about the parameters , here is what i got for the palmitoleic acid modified lysine but i am unsure about 2 bonds value  :

[ moleculetype ]
LYSPALM 1

[ atoms ]
 1 P2   1 LYSPALM BB   1  0    
 2 SC3  1 LYSPALM SC1  2  0    
 3 P2   1 LYSPALM SC2  3  0
 4 C1   1 LYSPALM C1A  4  0
 5 C4h  1 LYSPALM D2A  5  0
 6 C1   1 LYSPALM C3A  6  0
 7 C1   1 LYSPALM C4A  7  0


[ bonds ]


 1  2 1 0.33 5000
 2  3 1 0.33 5000 # unsure 
 3  4 1 	  # unsure 
 4  5 1 0.47 3800
 5  6 1 0.47 3800
 6  7 1 0.47 3800
You can notice that the SQp bead in lysine turned into a P2 bead to link the palmotoleic chain
Do i need to run all-atoms simulation on this one particular residue in water to get the bonds or we could have access to the bonds values in another way ?

**Reply by pckroon** [reply]
I've not been involved with model development for some time, so I honestly don't know. Maybe someone else will chip in, but in the end the real question is, what do you need to convince reviewer number 2 that your simulation is reliable?

**Reply by fgrunewald** [reply]
@amiel-ab Yes you need to derive the bonded parameters from an all-atom simulation yourself. Ideally, you'd get these parameters directly from a lipidated protein simulation. However, the fragment approach is a good first approximation if the other one is too complicated. Check out [this tutorial](http://cgmartini.nl/index.php/2021-martini-online-workshop/tutorials/560-5-parametrization-of-a-new-small-molecule) for more information on parametrization.

**Reply by amiel-ab** [reply]
Thanks a lot @pckroon  and @fgrunewald

**Reply by amiel-ab** [reply]
@fgrunewald i tried exactly as the tutorial but was not able to get a CG structure 
When i run the martinize2 command i got it stops at this step : 
 martinize2 -ff-dir force_fields -map-dir mappings -f CyaA-100-palmito_center.pdb -x CyaA_palm.pdb -o topol.top
    INFO - general - Read 1 molecules from PDB file CyaA-100-palmito_center.pdb
    INFO - step - Guessing the bonds.
    INFO - general - 1 molecules after guessing bonds
    INFO - step - Repairing the graph.
    INFO - general - Applying modification N-ter to residue A-MET1
Killed
You could try to run the same command as in the error message.
Do you see any obvious error ? 
Thank you for your patience.
[CyaA_lys_palmitoleil.zip](https://github.com/marrink-lab/polyply_1.0/files/15310753/CyaA_lys_palmitoleil.zip)

**Reply by pckroon** [reply]
You need to rename the residue with PALM tail in the PDB file to have the same name as the residue in the `lyspalm_charm.ff` file.  IIRC the PDB standard is limited to 3 characters for a residue name, so maybe LSP?

**Reply by amiel-ab** [reply]
I have changed to LSP but it does not seem to work, and give same error as previous

[Cya_LSP.zip](https://github.com/marrink-lab/polyply_1.0/files/15319665/Cya_LSP.zip)

**Reply by pckroon** [reply]
Could you run with `-vv` to see where it breaks?
As a side note, I transferred the discussion to the vermouth-martinize repo, since this is the more appropriate place.

**Reply by amiel-ab** [reply]
Here is the detailed version with -vv option:
DEBUG - step - vermouth.processors.repair_graph - Matching residue A-LSP860 to its reference
It stays at this step

**Reply by pckroon** [reply]
You made an interesting edge case :)
The atom names between your pdb file and charmm .ff file don't just not match, but the match wrong. In particular, C10-16 are Wrong. This causes a *dramatic* increase in the computational complexity of overlapping the residues. 
This extensive mismatch in names also causes issues earlier on, when guessing which atoms are connected in the PDB file.

Correcting some (or all) of those names should help. Correcting the names of the carbon atoms /should/ be enough. Note that your .ff file doesn't have a CD atom.

PS. I also spotted problems with your mapping file:
```
 Level 5 - inconsistent-data - vermouth.map_input - Can't find atom HE1 in block LSP in force field charmm
 Level 5 - inconsistent-data - vermouth.map_input - Can't find atom HE2 in block LSP in force field charmm
 Level 5 - inconsistent-data - vermouth.map_input - Can't find atom NZ in block LSP in force field charmm
 Level 5 - inconsistent-data - vermouth.map_input - Can't find atom H10A in block LSP in force field charmm
```

**Reply by amiel-ab** [reply]
Thanks, i have corrected some mistakes in mapping,changing the carbon names , adding C4 in my .ff files but it doesn't seem to work, i tried to change to hydrogen names in .ff to match those is PDB but it is a little tricky.
I have also noticed that the new P2 bead added in .ff file  for martini could not be assigned as SC1 so i had to change to SC2 but i am unsure about this.
[CyaA_martinize_files.zip](https://github.com/marrink-lab/vermouth-martinize/files/15336305/CyaA_martinize_files.zip)

**Reply by pckroon** [reply]
The number of carbon atoms in your ff/pdb files still don't match. C17 is the last carbon atom in your PDB file, but your ff file goes up to C16...
Please reduce your testpdb to just 1 LSP residue, and check your ff file carefully.  Focus on the carbon skeleton, the hydrogens are not as important

**Reply by amiel-ab** [reply]
Hi, 

My bad Peter, this time i took my time to correct every file but i am unable to see where i am making a mistake; if you are able to see it please do let me know.
[CyaA_lyspalm_v3.zip](https://github.com/marrink-lab/vermouth-martinize/files/15391609/CyaA_lyspalm_v3.zip)

**Reply by pckroon** [reply]
You have a typo in your `charmm/lyspalm.ff:138`. It says `61 54` instead of `61 64`. In addition, in your mapping you map `N02` to `SC2`, rather than `N01`. The atomnames in the mapping file should match the names you specified in the atomistic ff.
Finally, I'm left with the following warning: ` WARNING - unknown-input - vermouth.processors.canonicalize_modifications - Could not identify the modifications for residues ['LSP860'], involving atoms ['6314-H18']`, which means the atom H18 probably shouldn't be in your pdb file.

**Reply by amiel-ab** [reply]
I had in fact a wrong version  i corrected it so there is only one hydrogen linked to C9 and C10 respectively; the program now goes to the end but i got an error saying the mapping is wrong (see error.log file here) . Using the -vv option i was not able to go all the way up to see what can of mistake it generates.


[CyaA_palm_v4.zip](https://github.com/marrink-lab/vermouth-martinize/files/15406274/CyaA_palm_v4.zip)

**Reply by pckroon** [reply]
It works for me? Did you rename your mapping file to `.map` and throw it in a `mappings` folder?

```
├── CyaA-100-scrwl-palmito-good.pdb
├── force_fields
│   ├── charmm
│   │   └── lyspalm.ff
│   └── martini3001
│       └── lys_cg.ff
└── mappings
    └── lyspalm.map
```
My invocation: `martinize2 -f CyaA-100-scrwl-palmito-good.pdb -ff-dir force_fields/ -map-dir mappings/ -o derp.top -x derp.pdb`

There's a small issue with your coordinates (I think). `Could not identify the modifications for residues ['GLY792', 'ASN793'], involving atoms ['5773-N01']`, N01 is probably clashing with GLY792 and ASN793.
`These hydrogen atoms are not covered by a mapping. This is not the best idea. ['7273-LSP983:H10B', '6296-LSP860:H10B', '19609-LSP983:H1E', '18715-LSP860:H1E', '19612-LSP983:H2E', '18718-LSP860:H2E']`. The names of the hydrogens between the atomistic ff and the mapping file are inconsistent (or the mapping doesn't describe all of them). The latter is not a major issue, but worth having a look at while you're at it.

**Reply by amiel-ab** [reply]
Ok it is working fine, Thank you very much Peter for your help and patience !

**Reply by amiel-ab** [reply]
for the link i thin k it worked ; just  to be sure in my case i have two residues LSP may i have one sep.ff and sep2.ff files ?

**Reply by pckroon** [reply]
Not sure I understand what you mean by that. In general there is no requirement to keep things in one file though.

**Reply by amiel-ab** [reply]
I mean i have two LSPs one that got GLY before and SER after and one that got GLY before and ARG after so this should need two sep.ff no ?

**Reply by pckroon** [reply]
I don't think so, since LSP will be LSP, right? And if  you define your links as connecting to $protein_residues you also need only 2 (or even 1) link.
